File: 251029/겹치지 않는 사각형의 넓이/area-of-non-overlapping-rectangle.py
arr = [ [0] * 2000 for i in range (2000)]

cur = 1000;



# 칸 채우기
for num in range (3) :
    x1, y1, x2, y2 = tuple(map(int, input().split()));
    x1 += cur;
    y1 += cur;
    x2 += cur;
    y2 += cur;
    for i in range (x1, x2) :

        for j in range (y1, y2) :
            if num != 2 :
                arr[i][j] = 1
            else :
                arr[i][j] = 0;


area = 0;

for i in range (0, 2000) :
    for j in range (0, 2000) :
        if arr[i][j] == 1:
            area+=1;

print(area)
            

    


# Please write your code here.

File: 251030/겹치지 않는 사각형의 넓이/area-of-non-overlapping-rectangle.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        // Please write your code here.

        int offset = 2000;

        int arr [][] = new int[offset][offset];

        for(int i = 0; i < 3; i++){
            int x1 = sc.nextInt() + offset;
            int y1 = sc.nextInt() + offset;
            int x2 = sc.nextInt() + offset;
            int y2 = sc.nextInt() + offset;

            for(int c = x1; c < x2; c++) {

                for(int r = y1; r < y2; r++){

                    if( i != 2 ) {
                        arr[c][r] = 1;
                    }
                    else {
                        arr[c][r] = 0;
                    }
                }

            }

        }

        int answer = 0;

        for(int i = 0; i < offset; i++){
            for(int j = 0; j < offset; j++){
                if(arr[i][j] == 1) answer +=1;
            }
        }

        System.out.println(answer);


    }
}